### ISSUE TITLE
[FEATURE] Support v2 volume delta replica rebuilding based on snapshot checksum: data plane and control plane

### ISSUE THREAD

**Issue by derekbit** [longhorn/longhorn#9488]
## Is your feature request related to a problem? Please describe (👍 if you like this request)

https://github.com/longhorn/longhorn/issues/5573

**Comment by shuo-wu** [comment]
YES. And we already knew that it was not quite stable during node reboot or network disconnection. Probably for now we just need to verify some regular scenarios like the test cases I mention above.

**Comment by roger-ryao** [comment]
**Verified on v1.9.0-dev-20250112 20250114**

- longhorn `v1.9.0-dev-20250112` https://github.com/longhorn/longhorn/commit/8e58220d6f72ca69744d4f73bfe5437791bf622c

**Verified on v1.8.0-rc3 20250114**

- longhorn `v1.8.0-rc3` https://github.com/longhorn/longhorn/commit/7a8d7d1e5688f547ad53010105c26b3282f57a64


**The test steps**
https://github.com/longhorn/longhorn/issues/9488#issuecomment-2560726556

**Result Passed**
